### PR TITLE
Fixes Caste death timers not working

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
Um, I forgot to re-implement the actual checking for the death timers, which is like, the whole point of it. Anyway, here you go, It's there, now.
## Changelog
:cl:
fix: Caste death timers now work, whoops.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/evolution.dm [hunk 1]
```diff
@@ -164,6 +164,12 @@
 	if(CHECK_BITFIELD(new_caste_flags, CASTE_CANNOT_EVOLVE_IN_CAPTIVITY) && isxenoresearcharea(get_area(src)))
 		to_chat(src, span_warning("Something in this place is isolating us from Queen Mother's psychic presence. We should leave before it's too late!"))
 		return
+	// Check if there is a death timer for this caste
+	if(new_caste_type.death_evolution_delay)
+		var/death_timer = hive.caste_death_timers[new_caste_type.caste_type_path]
+		if(death_timer && timeleft(death_timer) > 0)
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/evolution.dm:170]: Shouldn't the timer be null anyway if it's reached zero? I don't think you need the timeleft check here.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/evolution.dm (hunk 1 above)
Comment (Arturlang) [reply code/modules/mob/living/carbon/xenomorph/evolution.dm:170]: Was worried about timeleft runtiming if death timer was null but I can check it if you want

Comment (Arturlang) [reply code/modules/mob/living/carbon/xenomorph/evolution.dm:170]: Ah, looks like I misunderstood the question

Comment (Arturlang) [reply code/modules/mob/living/carbon/xenomorph/evolution.dm:170]: Tested your suggestion and it looks like it works

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**